### ISSUE TITLE
IP-Per-Container broken with Marathon-v1.3.0+

### ISSUE THREAD

**Issue by djosborne** [mesosphere/marathon-lb#313]
Latest Marathon changes make it [impossible to specify `discoveryInfo` and `ipAddress` fields](https://github.com/mesosphere/marathon/issues/4054). Instead, Marathon devs say to use `PortMappings`. 

Marathon-LB needs to update [where it expects this information](https://github.com/mesosphere/marathon-lb/blob/af45a72833de03ddb6db0f6fdb39506308ffef64/utils.py#L185) to be the new location.

Sort-of unrelated good news - if a user specifies a VHOST and `PortMappings.ContainerPort`, Marathon is now correctly allocating servicePorts in the PortMappings section. So [that issue in Marathon](https://github.com/mesosphere/marathon/issues/3636) is now effectively resolved.

I've been meaning to tackle these issues but now it seems less likely I'll get to them this week, so just wanted to start a log here of what's going on for the community.

**Comment by brndnmtthws** [comment]
It looks like the app state has changed quite a bit in Marathon 1.3. I'm probably going to rip out the existing behaviour (auto assigning, etc), and replace it with the new stuff, then mark MLB 1.4 as being compatible with Marathon 1.3+ only (for IP-per-task). Does that sound reasonable?

**Comment by brndnmtthws** [comment]
Actually, it might be slightly more complicated than I thought. Let me investigate further.

**Comment by brndnmtthws** [comment]
PTAL at #329.

**Comment by djosborne** [comment]
Woo! Thanks for this @brndnmtthws ! I tried testing this yesterday with Calico but ran into troubles with my `app.json` (which turned out to just be typos on my end). Did a runthrough today with the v1.4.0 release and it works with calico networked docker tasks (even using VHOSTS), which is awesome!

Unfortunately, I found it does not work with CNI tasks, but I think the reason is that Marathon still is not auto-allocating servicePorts to CNI tasks. So will be raising a follow up report over with the Marathon team.

Thanks again!
